File: gaia-sdk-javascript/src/graphql/request/type/DeletedSkillProvisionImpulse.ts

import {TenantKeyOne} from "./TenantKeyOne";

import VariableRegistry from "../../../api/VariableRegistry"
import {Uuid, ISO8601, Struct} from "../../GaiaClient";
import {Order} from "../enumeration/Order";
import {OrderByField} from "../enumeration/OrderByField";
import {EdgeOrderByField} from "../enumeration/EdgeOrderByField";
import {EdgeType} from "../enumeration/EdgeType";

/**
 * Impulse which indicates the result of a delete SkillProvision impulse
 */
export class DeletedSkillProvisionImpulse extends Array<(_:VariableRegistry) => string> {
public _typeName = "DeletedSkillProvisionImpulse";
    public id = () => { 
        this.push(_ => "id")
    };

    public data = (config: (_:TenantKeyOne) => void) => this.push((registry) => {
        const entity = new TenantKeyOne();
        config(entity);
        return "data { " + entity.render(registry) + " }";
    });

    public render = (registry: VariableRegistry):String => this.map(e => e(registry)).join(" ");
}

File: gaia-sdk-python/gaia_sdk/graphql/response/type/DeletedSkillProvisionImpulse.py

from gaia_sdk.graphql.response.type.TenantKeyOne import TenantKeyOne

Uuid = str
String = str
ISO8601 = str
Struct = dict
Float = float
Int = int
Boolean = bool


class DeletedSkillProvisionImpulse:
    """
    Impulse which indicates the result of a delete SkillProvision impulse
    """
    dictionary: dict

    def __init__(self, dictionary: dict):
        self.dictionary = dictionary

    def __eq__(self, other):
        if type(other) is type(self):
            return self.dictionary == other.dictionary
        return False

    def __repr__(self):
        return {'dictionary': self.dictionary}

    @property
    def id(self) -> Uuid:
        return Uuid(self.dictionary.get("id"))
    @property
    def data(self) -> TenantKeyOne:
        return TenantKeyOne(self.dictionary.get("data"))
